java/com/google/location/lbs/activity/audioset/dataprocessing/CsvParser.java

```diff
@@ -0,0 +1,244 @@
+import java.io.File;
+import java.io.IOException;
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Arrays;
+
+/**
+ * CSV Class
+ *
+ * <p>CSV is a class for the reading of a dataset in a csv format.
+ *
+ * @author Carver Forbes
+ * @version 0.1
+ */
+public class CsvParser {
+
+  private File file;
+  private ArrayList<String> ytidList;
+  private HashMap<String, IncompleteExample.Example> idToProtoMap;
+  private HashMap<String, String> idToUrlMap;
+
+  private final boolean debug = true;
+
+  /**
+   * CSV constructor
+   *
+   * @param pathname the pathname for the csv file to be read
+   * @throws IllegalArgumentException if the pathname is incorrect
+   */
+  public CsvParser(String pathname) {
+    file = new File(pathname);
+    if (!file.exists()) {
+      throw new IllegalArgumentException(String.format("This file does not exist: %s", pathname));
+    }
+    ytidList = new ArrayList<>();
+    idToProtoMap = new HashMap<>();
+    idToUrlMap = new HashMap<>();
+  }
+
+  /** Tester Function to print 5 YouTube Links */
+  public void printYouTubeLinks() {
```

Comment:
add TODO to remove this temporary function -- ideally they should be converted to some unit tests if you want.

Reply:
Ok. I'll move it to the unit tests. Thanks!